workflow/scripts/featurecounts_deseq2.R

```diff
@@ -0,0 +1,382 @@
+
+            }
+            cat("\n",file=LogFile,append=TRUE,sep="")
+        # AVI: creates required output files with message
+        } else {
+            write("More than 2 samples treated with the same antibody are needed to calculate the FDR & LOGFC.", stdout())
+            outfiles <- c(snakemake@output[["FDR_1_perc_res"]],
+                          snakemake@output[["FDR_1_perc_bed"]],
+                          snakemake@output[["FDR_5_perc_res"]],
+                          snakemake@output[["FDR_5_perc_bed"]]
+                          )
+            for (f in outfiles) {
+                writeLines(paste0("More than 2 samples treated with the same antibody are needed to calculate the FDR & LOGFC."), f)
+            }
+        }
+
+        ## SAMPLE CORRELATION HEATMAP
+        pdf(file=snakemake@output[["plot_sample_corr_heatmap"]],width=10,height=8)  # AVI: added to create separate pdf files
+        rld.subset <- assay(rld)[,comp.samples]
+        sampleDists <- dist(t(rld.subset))
+        sampleDistMatrix <- as.matrix(sampleDists)
+        colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
+        pheatmap(sampleDistMatrix,clustering_distance_rows=sampleDists,clustering_distance_cols=sampleDists,col=colors)
+        dev.off()  # AVI: added to create separate pdf files
+
+        ## SCATTER PLOT FOR RLOG COUNTS
+        pdf(file=snakemake@output[["plot_scatter"]],width=10,height=8)  # AVI: added to create separate pdf files
+        combs <- combn(comp.samples,2,simplify=FALSE)
+        clabels <- sapply(combs,function(x){paste(x,collapse=' & ')})
+        plotdat <- data.frame(x=unlist(lapply(combs, function(x){rld.subset[, x[1] ]})),y=unlist(lapply(combs, function(y){rld.subset[, y[2] ]})),comp=rep(clabels, each=nrow(rld.subset)))
+        plot <- xyplot(y~x|comp,plotdat,
+                       panel=function(...){
+                           panel.xyplot(...)
+                           panel.abline(0,1,col="red")
+                       },
+                       par.strip.text=list(cex=0.5))
+        print(plot)
+        dev.off()  # AVI: added to create separate pdf files
+
+        colnames(comp.df) <- paste(CompPrefix,".",colnames(comp.df),sep="")
+        deseq2_results_list[[idx]] <- comp.df
+
+    }
+
+    ## WRITE RESULTS FROM ALL COMPARISONS TO FILE
+    deseq2_results_table <- cbind(interval.table,do.call(cbind, deseq2_results_list),raw.counts,pseudo.counts)
+    write.table(deseq2_results_table, file=ResultsFile, col.names=TRUE, row.names=FALSE, sep='\t', quote=FALSE)
+
+}
+
+#################################################
+#################################################
+### R SESSION INFO                             ##
+#################################################
+#################################################
+
+RLogFile <- snakemake@log[[1]]  # AVI: adapted to snakemake
+if (file.exists(RLogFile) == FALSE) {
```

Comment:
The `RLogFile` file path always exists, because it is created above as `LogFile` (line 222). So I guess we would just skip the `if` clause here and use `cat()` to append the `sessionInfo()`?